forem/forem
# Implement TwitchTV Unified Embed

## What type of PR is this? (check all applicable)


## Description
This PR implements the [unified embed tag](https://github.com/forem/forem/issues/15099) for Twitch embeds.

## Related Tickets & Documents
closes https://github.com/forem/forem/issues/15582

## QA Instructions, Screenshots, Recordings

Example Twitch URLs:
https://clips.twitch.tv/embed?clip=SpeedyVivaciousDolphinKappaRoss-IQl5YslMAGKbMOGM&parent=www.example.com
https://player.twitch.tv/?video=1222841752&parent=www.example.com
https://player.twitch.tv/?video=1222841752
https://www.twitch.tv/videos/1250164963
https://www.twitch.tv/monchi_tv/clip/CrepuscularSparklingGalagoBudBlast-ij3jvc4r437D4L4L
Twitch Video ID: 1222841752
Twitch Clip Slug: CrepuscularSparklingGalagoBudBlast-ij3jvc4r437D4L4L

- As a user, create a Twitch liquid tag using this format: `{% embed <twitch-url> %}`. Please use the example URLs provided above. The Twitch Liquid Tag should render properly.
- Also create Twitch liquid tag using the old method: `{% twitch <twitch-video-id> %}` or `{% twitch <twitch-clip-slug> %}`. This should also work, as I am leaving the old implementation in place for now.

### UI accessibility concerns?
None

## Added/updated tests?


## [Forem core team only] How will this change be communicated?
Considering the project as a whole, I shall update the Liquid Tag documentation at the end. Since I am not changing the current implementation, the present documentation is still valid. Plus, I will avoid updating the docs piecemeal, and risk confusing users.
